torch/nn/functional.py

```diff
@@ -547,6 +547,22 @@ def nll_loss(input, target, weight=None, size_average=True):
     return f(input, target)
 
 
+def poisson_nll_loss(log_input, target, size_average=True):
+    r"""Poisson negative log likelihood loss.
+
+    Drops :math:`log(target!)` term. See :class:`~torch.nn.PoissonNLLLoss` for details.
+
+    Args:
+        log_input: :math:`log(input)`
+        target: :math:`target \sim Pois(input)`
+    """
+    loss = torch.exp(log_input) - target * log_input
```

Comment:
The loss accepts log input to avoid cases of application log to non-positive numbers. Examples: [Article](http://www.mathstat.dal.ca/~hgu/Neural%20Comput%20&%20Applic.pdf)